tests/util/test_file.cpp

```diff
@@ -54,33 +52,53 @@ using namespace realm;
 
 TestFile::TestFile()
 {
-    static std::string tmpdir = [] {
-        disable_sync_to_disk();
-
-        const char* dir = getenv("TMPDIR");
-        if (dir && *dir)
-            return dir;
-#if REALM_ANDROID
-        return "/data/local/tmp";
+#ifdef _WIN32
+	GUID guid = { 0 };
+	wchar_t guid_string[40] = { 0 };
+	CoCreateGuid(&guid);
+	StringFromGUID2(guid, guid_string, 40);
+	std::wstring filename = L"realm_testfile_" + std::wstring(guid_string).substr(1, 36) + L".realm";
+	path = (std::filesystem::temp_directory_path() / filename).u8string();
 #else
-        return "/tmp";
-#endif
-    }();
-    path = tmpdir + "/realm.XXXXXX";
+	static std::string tmpdir = tmp_dir();
+	path = util::File::resolve("realm.XXXXXX", tmpdir);
     int fd = mkstemp(&path[0]);
     if (fd < 0) {
         int err = errno;
         throw std::system_error(err, std::system_category());
     }
     close(fd);
     unlink(path.c_str());
+#endif
 
     schema_version = 0;
 }
 
 TestFile::~TestFile()
 {
+	if (in_memory)
+		return;
+
+#ifdef _WIN32
+	bool success = false;
+	int attempts = 0;
+
+	do {
+		try {
+			util::File::try_remove(path);
+			success = true;
+		}
+		catch (const util::File::AccessError& e) {
+			attempts++;
+			std::this_thread::sleep_for(std::chrono::milliseconds(15));
+		}
+	} while (!success && attempts < 3);
+
+	if (!success)
+		printf("Failed to delete realm file at %s\n", path.c_str());
```

Comment:
`printf()`? Isn't `std::cerr` better?

Reply:
That’s diagnostic code I’ll need to delete before merging. It helped me diagnose that cleaning up test realms on Windows doesn’t work because they’re still in use by sync when the `TestFile` destructor is invoked.